project/webapp/controller/ProductsList.controller.js

```diff
@@ -186,18 +190,68 @@ sap.ui.define(
               })
             );
           }
-          if (suppliersFilter) {
+          
             aFilters.push(
               new Filter({
                 filters: suppliersFilter,
                 and: false
               })
             );
-          }
+          
 
           return aFilters;
         },
-      }
-    );
+
+        onSearchProducts(oEvent) {
+          const oTableBinding = this.byId("productList").getBinding("items");
+          const aFilters = this._getAllFilters(oEvent);
+
+          oTableBinding.filter(aFilters);
+        },
+
+        onClearFilters() {
+          const oTableBinding = this.byId("productList").getBinding("items");
+
+          this.byId("releaseDate").setValue(null);
+          this.byId("supplier").setTokens([]);
+          this.byId("searchName").setValue(null);
+          this.byId("categorySelect").setSelectedKeys([]);
+
+          oTableBinding.filter(null);
+        },
+
+        _deleteProduct() {
+          // TODO: add products delete functionality
+        },
+
+        onDeleteProductPress() {
+          MessageBox.confirm(this._getConfirmationText(), {
+            title: this.oResourceBundle.getText("ConfirmDeleteProductTitle"),
+            actions: [MessageBox.Action.OK, MessageBox.Action.CLOSE],
+            onClose: (sAction) => {
+              switch (sAction) {
+                case MessageBox.Action.OK:
+                  this._deleteProduct();
+                  break;
+
+                default:
+                  break;
+              }
+            },
+          });
+        },
+
+        _getConfirmationText() {
+          const aSelectedItems = this.byId("productList").getSelectedItems();
+          const sPath = aSelectedItems[0].getBindingContext(this.APP_MODEL_NAME).getPath();
+          const sProductName = aSelectedItems[0].getBindingContext(this.APP_MODEL_NAME).getModel().getProperty(`${sPath}/name`);
+
+          if (aSelectedItems.length > 1) {
+            return this.oResourceBundle.getText("ConfirmDeleteProductsText", [aSelectedItems.length]);
```

Comment:
you're twice wrote return this.oResourceBundle.getText("ConfirmDeleteProductsText"
to improve that use ternary operator:
return this.oResourceBundle.getText("ConfirmDeleteProductsText",
              [aSelectedItems.length > 1 ? aSelectedItems.length : sProductName]
          );